src/components/routes/RouteItemEdit.vue

```diff
@@ -0,0 +1,212 @@
+<script lang="ts">
+/**
+ * RouteItemEdit Component
+ *
+ * @description * Use this component to render an item in a list with the
+ * option to edit data.
+ *
+ * Note: This component is commonly used in `RouteListEdit`.
+ *
+ * @props
+ * - `route` (RouteItem, required): The object representing the route.
+ *   It should be of type `RouteItem`
+ *
+ * @example
+ * <route-list-item :route="route" />
+ *
+ * @see [Figma Design](https://www.figma.com/file/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?type=design&node-id=4858%3A104042&mode=dev)
+ */
+
+// libraries
+import { computed, defineComponent, ref } from 'vue';
+import { i18n } from 'src/boot/i18n';
+
+// composables
+import { useRoutes } from 'src/composables/useRoutes';
+
+// types
+import type { FormOption } from '../types/Form';
+import type { RouteItem } from '../types/Route';
+
+export default defineComponent({
+  name: 'RouteItemEdit',
+  props: {
+    route: {
+      type: Object as () => RouteItem,
+      required: true,
+    },
+  },
+  setup(props) {
+    const action = ref<string>('input-distance');
+    const distance = ref<number>(props.route.distance || 0);
+
+    const optionsAction: FormOption[] = [
+      {
+        label: i18n.global.t('routes.actionInputDistance'),
+        value: 'input-distance',
+      },
+    ];
+
+    const { getRouteIcon } = useRoutes();
+    const optionsTransport: FormOption[] = [
+      {
+        label: '',
+        value: 'bike',
+        icon: getRouteIcon('bike'),
+      },
+      {
+        label: '',
+        value: 'walk',
+        icon: getRouteIcon('walk'),
+      },
+      {
+        label: '',
+        value: 'bus',
+        icon: getRouteIcon('bus'),
+      },
+      {
+        label: '',
+        value: 'car',
+        icon: getRouteIcon('car'),
+      },
+      {
+        label: '',
+        value: 'none',
+        icon: getRouteIcon('none'),
+      },
+    ];
+
+    const transport = ref<string>(props.route.transport);
+    const isShownDistance = computed((): boolean => {
+      return (
+        transport.value === 'bike' ||
+        transport.value === 'walk' ||
+        transport.value === 'bus'
+      );
+    });
+
+    return {
+      action,
+      distance,
+      isShownDistance,
+      optionsAction,
+      optionsTransport,
+      transport,
+    };
+  },
+});
+</script>
+
+<template>
+  <div class="row items-center" data-cy="route-list-item">
+    <div class="col-12 col-sm-2" data-cy="column-direction">
+      <!-- Column: Direction -->
+      <div
+        class="flex gap-8 text-subtitle2 text-weight-bold text-grey-10"
+        data-cy="label-direction"
+      >
+        <!-- From work -->
+        <span v-if="route.direction === 'from_work'">
+          <q-icon
+            name="arrow_back"
+            size="18px"
```

Comment:
Size prop value should be const (value repeat on the code line 121).

Reply:
Refactored into a variable.